webapp/tests/Unit/Controller/Team/ProblemControllerTest.php

```diff
@@ -183,4 +184,40 @@ public function testInteractiveSamples(): void
         $response = $this->client->getResponse();
         self::assertEquals(404, $response->getStatusCode());
     }
+
+    /**
+     * Test that the problems page does not show page, statement, sample data before contest start.
+     */
+    public function testAccessProblemBeforeContestStarts(): void
+    {
+        /** @var EntityManagerInterface $em */
+        $em = self::getContainer()->get(EntityManagerInterface::class);
+
+        $contest = $em->getRepository(Contest::class)->findOneBy(['shortname' => 'demo']);
+        $originalStartTime = $contest->getStarttimeString();
+
+        $lastSpacePosition = strrpos($originalStartTime, ' ');
+        $datetime = substr($originalStartTime, 0, $lastSpacePosition);
+        $timezone = substr($originalStartTime, $lastSpacePosition + 1);
+
+        $date = new \DateTime($datetime, new \DateTimeZone($timezone));
+        $date->modify('+1 hour');
+        $newTimeString = $date->format('Y-m-d H:i:s') . ' ' . $timezone;
+        $contest->setStarttimeString($newTimeString)->updateTimes();
+        
+        $this->withChangedConfiguration('public_access_before_contest', true, function () {
+            $this->client->request('GET', '/public/problems');
+            static::assertSelectorTextContains('.nav-item .nav-link.disabled', 'Problems');
+            static::assertSelectorTextContains('.alert.alert-secondary', 'No problem texts available at this point.');
+
+            $this->client->request('GET', '/team/problems/2/statement');
```

Comment:
Can you do a loop here? And I wonder if `/team/problems/2` should also throw a `404` as it would disclose the number of problems already.

Reply:
I directly wrote a loop from one to three to test all three problems. Was this necessary, or is there somewhere that reveals the number of problems that I missed?

Reply:
You could ask the ContestProblems from the Contest and loop over that.

Reply:
Is it okay for me to do this now?

Reply:
I think its really better to get the problems from the contest.

Reply:
Dealt with it